app/validation.py

```diff
@@ -115,6 +125,22 @@ def validates_against_schema(validator_name, submitted_json):
         return True
 
 
+def get_validation_errors(validator_name, json_data, enforce_required=True):
+    error_map = {}
+    try:
+        validator = get_validator(validator_name, enforce_required)
+        errors = sorted(validator.iter_errors(json_data), key=lambda e: e.path)
+        for index, error in enumerate(errors):
+            if error.path:
+                key = error.path.pop()
+            else:
+                key = "err_{}".format(index)
```

Comment:
> How are we going to tie the errors to individual fields for display?
The keys here (from `error.path`) are the individual field names that are in error.  The frontend apps will be responsible for mapping these keys to the message to be displayed to the user.  The plan is that each field will only have a single error message, so we don't really care if there are multiple errors on the same field - the same error message will be displayed regardless. 

It's only "form errors" that will get the 'err_x' key - these are effectively a special key for form errors.